Improvement/zenko 4871 readonly for pra

KillianG opened the pull request.

Commit:
Test readonly on DR site

Issue: ZENKO-4871

Commit:
Test with assume role with web identity for data accessor

Issue: ZENKO-4871

bert-e commented:
# Hello killiang,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4871 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.66`

* `2.7.62`

* `2.8.42`

* `2.9.17`


Please check the `Fix Version/s` of ZENKO-4871, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

KillianG commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.10/improvement/ZENKO-4871-readonly-for-pra` with contents from `w/2.9/improvement/ZENKO-4871-readonly-for-pra`
and `development/2.10`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.10/improvement/ZENKO-4871-readonly-for-pra origin/development/2.10
 $ git merge origin/w/2.9/improvement/ZENKO-4871-readonly-for-pra
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.10/improvement/ZENKO-4871-readonly-for-pra
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #4c2ee0add1d3f605ada5da7355232ecc183088c2 on the integration branch
`w/2.7/improvement/ZENKO-4871-readonly-for-pra` is merging a branch which is neither the current
branch `improvement/ZENKO-4871-readonly-for-pra` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/ZENKO-4871-readonly-for-pra` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

Commit:
Precise that the put object should be rejected for DATA_ACCESSOR

Issue: ZENKO-4871

KillianG commented:
/reset

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

KillianG commented:
/create_integration_branches

bert-e commented:
# History mismatch



Merge commit #4c2ee0add1d3f605ada5da7355232ecc183088c2 on the integration branch
`w/2.7/improvement/ZENKO-4871-readonly-for-pra` is merging a branch which is neither the current
branch `improvement/ZENKO-4871-readonly-for-pra` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/ZENKO-4871-readonly-for-pra` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

KillianG commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.10/improvement/ZENKO-4871-readonly-for-pra` with contents from `w/2.9/improvement/ZENKO-4871-readonly-for-pra`
and `development/2.10`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.10/improvement/ZENKO-4871-readonly-for-pra origin/development/2.10
 $ git merge origin/w/2.9/improvement/ZENKO-4871-readonly-for-pra
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.10/improvement/ZENKO-4871-readonly-for-pra
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

williamlardier commented on tests/ctst/features/pra.feature:
```diff
@@ -22,6 +22,10 @@ Feature: PRA operations
     # Check that objects are transitioned in the DR site
     Given access keys for the replicated account
+
+    When the DATA_ACCESSOR user tries to perform PutObject on "DR" site
+    Then it "should not" pass Vault authentication
```
Are we sure this won't be flaky? As we just deployed DR at this stage, the "access keys for the replicated account" may complete quickly with no guarantee that the data accessor was already replicated
While having this check after the checks for the transitioned objects, we benefit from the waiting times

KillianG replied:
If drsink and drsource are running everything is replicated

KillianG replied:
we do the check after the drsink and drsource are in running phase, which means everything is replicated, if something is not then its a bug and we want the test to fail

williamlardier replied:
ye sok replicated through the initial dump, but not sure this is enough if the AK was created after (it may need time to replicated)
But didn't you get an issue with this test at first, and then it worked?

KillianG replied:
AK are created on the dr site itself, i am not sure i understand what you want to be replicated exactly ?
Never had such an issue AFAIR

williamlardier replied:
Sorry I meant the role, not AK; the role will need to be replicated at the time this step runs, and having the step "running" is not enough iff the role was created after the dump, but that shouldn't be the case

williamlardier commented on tests/ctst/steps/pra.ts:
```diff
@@ -274,6 +275,26 @@ Then('object {string} should {string} be {string} and have the storage class {st
+
+    await this.setupEntity(EntityType.DATA_ACCESSOR);
+    
+    const preName = this.getSaved<string>('accountName') ||
+        this.parameters.AccountName || Constants.ACCOUNT_NAME;
```
Code not used

```suggestion
```

williamlardier commented on tests/ctst/steps/pra.ts:
```diff
@@ -274,6 +275,26 @@ Then('object {string} should {string} be {string} and have the storage class {st
+        Identity.useIdentity(IdentityEnum.ACCOUNT, Zenko.sites['source'].accountName);
+    }
+    this.resetCommand();
+    this.addToSaved('accountName', Zenko.sites['source'].accountName);
```
shouldn't this be in the if/else above, so we use the -replicated one if we are using the data_accessor role from this account?
If we want to use this for the later tests, we should put this line at the end of the step

KillianG replied:
The account name in zenkos is always the same, the `-replicated` is only used in the test to determine the account we are using, but in the end the account name is the same

KillianG replied:
that said im not sure i understand this
> If we want to use this for the later tests, we should put this line at the end of the step

francoisferrand commented on .github/scripts/end2end/deploy-zenko.sh:
```diff
@@ -37,7 +37,8 @@ else
 
 # TODO: use kustomize
-export ZENKO_ANNOTATIONS="annotations:"
+export ZENKO_ANNOTATIONS="annotations:
```
should not be added manually : it will be added automatically by the DrCtl

francoisferrand commented on tests/ctst/steps/pra.ts:
```diff
@@ -163,7 +164,7 @@ async function waitForPhase(
 }
 
-const installTimeout = 360000;
+const installTimeout = 600000;
```
why is it increasing?
is it there really an issue with the duration here?

williamlardier replied:
I got the same issue on my build, reviewing the logs showed no error, it was just slower than 6min

KillianG replied:
I am testing some things for the flaky tests, this is not related to this ticket exactly, just trying to make things less flaky..

francoisferrand replied:
> I am testing some things for the flaky tests, this is not related to this ticket exactly, just trying to make things less flaky..

please do that in another branch (even if based on top of this one) : try to keep things separate, and one ticket = one PR (esp. when "testing the CI", it creates lots of noise to keep pushing to the PR : prefer to use a branch which is not in an open PR, PR are for _review_ before _merging_, not testing...)

KillianG replied:
ok

francoisferrand commented on .github/scripts/end2end/configs/zenko.yaml:
```diff
@@ -52,7 +52,7 @@ spec:
         storageClassName: "standard"
   kafkaCleaner:
-    interval: 30s
+    interval: 2m
```
is it really needed?

francoisferrand commented on .github/workflows/end2end.yaml:
```diff
@@ -487,6 +487,9 @@ jobs:
         working-directory: ./.github/scripts/end2end
+      - name: Debug wait
+        uses: ./.github/actions/debug-wait
+        timeout-minutes: 60
```
not to be commited

francoisferrand commented on solution/deps.yaml:
```diff
@@ -21,7 +21,7 @@ cloudserver:
   sourceRegistry: ghcr.io/scality
   image: zenko-drctl
-  tag: v1.0.4
+  tag: a38d1f8848d8422317584effcbcf0fa2b1de04d9
```
not to be commited

bert-e commented:
# History mismatch



Merge commit #856eed78e9e978c9bb8fbe87db545e219c8d3af7 on the integration branch
`w/2.7/improvement/ZENKO-4871-readonly-for-pra` is merging a branch which is neither the current
branch `improvement/ZENKO-4871-readonly-for-pra` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/ZENKO-4871-readonly-for-pra` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

Commit:
Remove useless annotation

Issue: ZENKO-4871

KillianG commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.10/improvement/ZENKO-4871-readonly-for-pra` with contents from `w/2.9/improvement/ZENKO-4871-readonly-for-pra`
and `development/2.10`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.10/improvement/ZENKO-4871-readonly-for-pra origin/development/2.10
 $ git merge origin/w/2.9/improvement/ZENKO-4871-readonly-for-pra
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.10/improvement/ZENKO-4871-readonly-for-pra
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

Commit:
Add base for annotation

Issue: ZENKO-4871

williamlardier reviewed (approved):
lgtm (I know we have flakies, but unrelated to these changes...)

Commit:
Slightly increase kfkc interval to make dr on restore work

Issue: ZENKO-4871

Commit:
Change step position for readonly check to gain some delay before testing

Issue: ZENKO-4871

Commit:
Launch ensure vault seed job to be sure data accessor role is created

Issue: ZENKO-4871

Commit:
Fix lint issue

Issue: ZENKO-4871

Commit:
chore: Update step position for readonly check to delay testing

Issue: ZENKO-4871

francoisferrand commented on tests/ctst/steps/cronjobs/countItems.ts:
```diff
@@ -2,12 +2,30 @@ import { When } from '@cucumber/cucumber';
+        await createJobAndWaitForCompletion(this, `${zenkoName}-ops-count-items`);
+        break;
+    case 'EnsureVaultSeeds':
+        await createJobAndWaitForCompletion(this, `${zenkoName}-ops-vault-check-seeds`);
```
instead of having a new list of jobs, may be better to just specify the actual name (or suffix) of the job in argument:

```suggestion
    await createJobAndWaitForCompletion(this, `${zenkoName}-ops-${job}`);
```

and call it accordingly: `when the count-items cronjobs completes without error`

(less code, less maintenance, less mistake because we give yet another name to the same thing. the benefit of such an "enum" would be to have compile-time validation, which is not the case here...)

KillianG commented:
/wait

KillianG commented:
/approve

Seragonia committed:
Ensure data accessor is propagated in DR site

As the data accessor is seeded by vault's cronjob, it might take
some time to be processed by kafka, so accessing it directly
won't work.

Issue: ZENKO-4871

Seragonia committed:
Generalize cronjob step logic

Issue: ZENKO-4871

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/Zenko/commit/7d651391e74df2375233df972c92782dc0cc07db) did not succeed in branch w/2.7/improvement/ZENKO-4871-readonly-for-pra.




*The following options are set:* **approve, create_integration_branches**
